src/Shared/TaskLoggingHelper.cs

```diff
@@ -279,6 +290,10 @@ public void LogMessage(MessageImportance importance, string message, params obje
                 ResourceUtilities.FormatString(message, messageArgs);
             }
 #endif
+            if (!LogsMessagesOfImportance(importance))
```

Comment:
Why does this (and equivalent) have to be after the checks?

Reply:
I didn't like the idea of parameters being validated conditionally. The caller should be getting the same behavior regardless of whether the log output was swallowed or not. I understand it is at the expense of performance but the impact is negligible and it prevent bugs where something breaks only when building at high enough log verbosity.

Reply:
That's reasonable. I was thinking that this only partially validates the input anyway, which might save some people without saving everyone, which could be confusing, but the more errors prevented the better.